[core-kit] sys-devel/gnuconfig Autogen gnuconfig package and remove it from core-kit package lists

org-tekeli-borisp opened the pull request:
Introduce a new template for the gnuconfig package with essential build and install functions. Include autogen.yaml to define directory-based package generation for fetching and updating config files. Remove gnuconfig from gentoo-staging

(cherry picked from commit 8cb75aa9fb246776e58ce9d2c39f778181d18f88) (cherry picked from commit beab382a2e1b6cb40545faed825fb30173e4e23a) (cherry picked from commit 6d213203690fe3471a5651426ee88c9161ea4770) (cherry picked from commit 6a7967861573d9873172869ea11b2b7b30078df5)

```
pdating non-funtoo repositories...
Sync successful and kits in alignment! :)

These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild     U  ] sys-devel/gnuconfig-20240728::core-kit [20180101::core-kit] 0 KiB

Total: 1 package (1 upgrade), Size of downloads: 0 KiB

Would you like to merge these packages? [Yes/No] 
>>> Verifying ebuild manifests
>>> Emerging (1 of 1) sys-devel/gnuconfig-20240728::core-kit
>>> Installing (1 of 1) sys-devel/gnuconfig-20240728::core-kit
>>> Jobs: 1 of 1 complete                           Load avg: 1.38, 1.91, 1.93
>>> Auto-cleaning packages...

>>> No outdated packages were found on your system.
```

org-tekeli-borisp committed:
[core-kit] sys-devel/gnuconfig Autogen gnuconfig package and remove it from core-kit package lists

Introduce a new template for the gnuconfig package with essential build and install functions. Include autogen.yaml to define directory-based package generation for fetching and updating config files. Remove gnuconfig from gentoo-staging

(cherry picked from commit 8cb75aa9fb246776e58ce9d2c39f778181d18f88)
(cherry picked from commit beab382a2e1b6cb40545faed825fb30173e4e23a)
(cherry picked from commit 6d213203690fe3471a5651426ee88c9161ea4770)
(cherry picked from commit 6a7967861573d9873172869ea11b2b7b30078df5)

geaaru commented on core-kit/next/sys-devel/gnuconfig/templates/gnuconfig.tmpl:
```diff
@@ -0,0 +1,37 @@
+
+src_compile() { :;}
+
+src_test() {
```
This could be removed considering that the USE flag `test` is not present in IUSE.

geaaru commented on core-kit/next/sys-devel/gnuconfig/templates/gnuconfig.tmpl:
```diff
@@ -0,0 +1,37 @@
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="5"
```
I think that you can set EASI=7 and drop `src_unpack` (just test it to validate that unpack phase is correctly executed)

org-tekeli-borisp replied:
done!

org-tekeli-borisp committed:
[core-kit] sys-devel/gnuconfig Update gnuconfig template to use EAPI 7 and streamline phases

Switch gnuconfig template to EAPI 7 and simplify src phases by removing unnecessary unpack, test, and modified implementation logic. This update aligns with modern practices and improves maintainability.

(cherry picked from commit efd4ed7afd12ae2b18b9c8b4eb8d8fdc37fe6665)
(cherry picked from commit e692660b3ffb9a85ead3db408034a8758aab6cc1)
